packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1474,10 +1475,34 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     }
   }
 
+  /// Collected during [describeSemanticsConfiguration], used by
+  /// [assembleSemanticsNode] and [_combineSemanticsInfo].
+  List<InlineSpanSemanticsInformation>? _semanticsInfo;
+
+  // Caches [SemanticsNode]s created during [assembleSemanticsNode] so they
+  // can be re-used when [assembleSemanticsNode] is called again. This ensures
+  // stable ids for the [SemanticsNode]s of [TextSpan]s across
+  // [assembleSemanticsNode] invocations.
+  Queue<SemanticsNode>? _cachedChildNodes;
+
   @override
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
-
+    _semanticsInfo = _textPainter.text!.getSemanticsInformation();
+    // TODO(chunhtai): the macOS does not provide a public API to support text
+    // selections across multiple semantics nodes. Remove this platform check
+    // once we can support it.
+    // https://github.com/flutter/flutter/issues/77957
```

Comment:
I have to make a fairly ugly compromise which i think we don't have a choice at the moment.

Reply:
It seems like this is the right choice because it would be a much bigger change to get it working like on native Mac, but let me know if I understand correctly:

So selecting some regular text and a link together would be an example of the problem, right?

![Screen Shot 2021-03-11 at 12 26 06 PM](https://user-images.githubusercontent.com/389558/110850770-0a3cda00-8265-11eb-843d-400f09936359.png)

On most platforms, this situation can't happen, because they can't select text using a screen reader?  What happens on native Mac in this case?

Reply:
Yes selecting regular text and link will be a problem

for screen reader in macOS, you use keyboard to change your accessibility focus. your mouse work as usual, so you can still select text and screen reader announce your selections. I am not entirely sure whether this is the same case for windows so I only disable macOS for now.

The native behavior is quite strange, it look as if it created a semantics object just for the selection. See the black box in the picture.

![Screen Shot 2021-03-10 at 5 06 25 PM](https://user-images.githubusercontent.com/47866232/110852460-440ee000-8267-11eb-9aa9-b9360b1770c5.png)

I talked to the chromium team about how do they handle this. Their reply is they just uses bunch of private API, and the system will generate those highlight for you. We can't use those private API because it will result in app store rejection.

We can probably hack something in and create the semantics object on our own if there is a selection, but I would like to wait until we finish Windows/Linux part first to get the full picture.

Reply:
Sounds good, thanks for the explanation!  I'm curious how Windows and Linux do this as well.  Maybe we can create something that works for all 3 instead of needing to exactly copy Windows.